From 539663626b91567159a70791c0f2b3b167135e9b Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Thu, 16 Jan 2025 15:50:46 +0100
Subject: [PATCH] Fix "Fix compilation with uclibc"

The ICU code must initialize the iconv members as well.

Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
Upstream: https://gitlab.gnome.org/GNOME/libxml2/-/commit/539663626b91567159a70791c0f2b3b167135e9b
---
 encoding.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/encoding.c b/encoding.c
index 41ecde18..0ce8d479 100644
--- a/encoding.c
+++ b/encoding.c
@@ -1641,6 +1641,10 @@ xmlCreateUconvHandler(const char *name, xmlCharEncodingHandler **out) {
     }
     enc->input = NULL;
     enc->output = NULL;
+#ifdef LIBXML_ICONV_ENABLED
+    enc->iconv_in = (iconv_t) -1;
+    enc->iconv_out = (iconv_t) -1;
+#endif
     enc->uconv_in = ucv_in;
     enc->uconv_out = ucv_out;
 
-- 
2.34.1

